src/BaGet.Core/Search/DatabaseSearchService.cs

```diff
@@ -178,32 +178,39 @@ public Task IndexAsync(Package package, CancellationToken cancellationToken)
             var frameworks = GetCompatibleFrameworksOrNull(framework);
             var search = (IQueryable<Package>)_context.Packages.Where(p => p.Listed);
 
-            if (!string.IsNullOrEmpty(query))
+            IQueryable<Package> ComposePackageQuery(IQueryable<Package> packageQuery)
             {
```

Comment:
We could add `packageQuery = packageQuery.Where(p => p.Listed);` here.

This lets you simplify line 179 to:

```csharp
var search = (IQueryable<Package>)_context.Packages;
```

And lines 237-239 to:

```csharp
var results = await AddSearchFilters(search).ToListAsync(cancellationToken);
```

Reply:
I would like to change/update `search` before using it. Maybe someone later will add some more lines of code, and I would like to have this filtering in that search already. Please let me know if you agree with it.